components/brave_perf_predictor/browser/perf_predictor_tab_helper.cc

```diff
@@ -0,0 +1,144 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_perf_predictor/browser/perf_predictor_tab_helper.h"
+
+#include "brave/components/brave_perf_predictor/browser/third_parties.h"
+#include "brave/components/brave_perf_predictor/browser/third_party_extractor.h"
+#include "brave/components/brave_perf_predictor/common/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+namespace brave_perf_predictor {
+
+PerfPredictorTabHelper::PerfPredictorTabHelper(
+    content::WebContents* web_contents)
+    : WebContentsObserver(web_contents) {
+  VLOG(3) << "Created PerfPredictorTabHelper";
+  if (web_contents->GetBrowserContext()->IsOffTheRecord()) {
+    return;
+  }
+
+  ThirdPartyExtractor* extractor = ThirdPartyExtractor::GetInstance();
+  if (!extractor->is_initialized()) {
+    extractor->load_entities(static_third_party_config);
```

Comment:
if this is static can't it just load the entities itself when it's created? Or is there something you're trying to decouple here?